src/components/routes/saved-routes/saved-routes.tsx

```diff
@@ -76,24 +78,19 @@ class SavedRoutes extends React.Component<Props, State> {
       return;
     }
 
-    const userRoutes = savedRoutes.map(route => {
+    const existingRoutes = savedRoutes.map(route => {
       if (route.locationFromName && route.locationToName) {
         return route;
       }
-    }) as Route[];
+    }).filter( (route: Route | undefined): route is Route => !!route );
 
-    const shortRoutes = userRoutes.splice(0, 2);
-    const routes = showAllUserRoutes ? userRoutes : shortRoutes;
+    const shortRoutes = existingRoutes.splice(0, 2);
+    const routes = showAllUserRoutes ? existingRoutes : shortRoutes;
```

Comment:
if showAllUserRoutes is true, shortRoutes is unused
```suggestion
    const routes = showAllUserRoutes ? existingRoutes : existingRoutes.splice(0, 2);
```